### PR TITLE
Introduce blackboard support

### PR DESCRIPTION
**Pull request by MarcoLm993**
The Blackboard support is up and running, together with some very basic documentation about it!

In the current state, we expect that BlackBoard variables are accessed only inside the state body (hence Transitions and related subclasses), and not in the state's `onentry` and `onexit` blocks (we can think about removing this limitation later, but for now I think it is fair).
As usual, tests for this new feature have been added and some bugs have been spotted and solved :)

### CODE REVIEW

**Comment by ct2034** [comment]
Great work. I will definitely give it a try!

File: docs/source/scxml-jani-conversion.rst [hunk 1]
```diff
@@ -33,14 +33,68 @@ Low-Level SCXML Conversion
...
-TODO: Describe how we translate the high-level SCXML to the low-level SCXML.
+Handling of (ROS) Timers
+__________________________
+
+TODO
+
+Handling of (ROS) Services
+_____________________________
+
+ROS services, as well as ROS topics, can be handled directly in the ROS to plain SCXML conversion, without the need of adding JANI-specific features, as for the ROS timers.
```
Comment (MKlauck) [inline docs/source/scxml-jani-conversion.rst:50]: In the beginning above we only talk about high and low level SCXML but here also the terminology of plain SCXML is used...

File: docs/source/scxml-jani-conversion.rst [hunk 2]
```diff
@@ -33,14 +33,68 @@ Low-Level SCXML Conversion
...
+
+.. image:: graphics/ros_action_to_scxml.drawio.svg
+    :alt: Handling of ROS Actions
+    :align: center
+
+
+Handling the BT Blackboard
+_____________________________
+
+The Blackboard is a container that shares variables across different BT plugins. The value if those variables normally changes over time, and is expected to be updated at each tick.
```
Comment (MKlauck) [inline docs/source/scxml-jani-conversion.rst:76]: Typo: the value of those variables

**Review by MKlauck — COMMENTED** [review]
Thank you very much for all the effort and for finishing this before your vacation :+1: I guess Christian will have a closer and more detailed look into the code specifics. From my side it already looks great.

File: docs/source/scxml-jani-conversion.rst (hunk 1 above)
Comment (MarcoLm993) [reply docs/source/scxml-jani-conversion.rst]: I gave it a pass, feel free to look at that again (resolving this comment in the meanwhile)

File: docs/source/scxml-jani-conversion.rst (hunk 2 above)
Comment (MarcoLm993) [reply docs/source/scxml-jani-conversion.rst]: Fixed

File: docs/source/scxml-jani-conversion.rst [hunk 3]
```diff
@@ -6,41 +6,99 @@ SCXML and JANI
 
 In CONVINCE, we expect developers to use Behavior Trees and SCXML to model the different parts of a robotic systems.
 
-SCXML (Scope XML) is a high level format that describes a single state machine, and allows it to exchange information with other state machines using events. Each SCXML file defines its variables (datamodel), states, and transitions.
+SCXML (Scope XML) is an XML format that describes a single state machine, and allows it to exchange information with other SCXML state machines using events. Each SCXML file defines its variables (datamodel), states, and transitions.
```
Comment (ct2034) [inline docs/source/scxml-jani-conversion.rst]:
```suggestion
SCXML (State Chart XML) is an XML format that describes a single state machine, and allows it to exchange information with other SCXML state machines using events. Each SCXML file defines its variables (datamodel), states, and transitions.
```

File: src/as2fm/jani_generator/jani_entries/jani_assignment.py [hunk 4]
```diff
@@ -30,12 +30,23 @@ class JaniAssignment:
...
+    def get_target(self):
+        """Return the variable storing the expression result."""
+        return self._var_name
+
+    def get_expression(self):
+        """Return the expression assigned to the target variable (or array entry)"""
+        return self._value
+
+    def get_index(self) -> int:
+        return self._index
```
Comment (ct2034) [inline src/as2fm/jani_generator/jani_entries/jani_assignment.py:48]:
```suggestion
        """Returns the index, i.e. the number that defines the order of execution in Jani."""
        return self._index
```

File: src/as2fm/scxml_converter/bt_converter.py [hunk 5]
```diff
@@ -24,19 +24,92 @@
...
+            if is_blackboard_reference(p_value):
+                var_name = get_blackboard_variable_name(p_value)
+                existing_bt_type = blackboard_vars.get(var_name)
+                assert existing_bt_type is None or existing_bt_type == p_type
+                blackboard_vars.update({var_name: p_type})
+    return blackboard_vars
+
+
+def generate_blackboard_scxml(bt_blackboard_vars: Dict[str, str]) -> ScxmlRoot:
+    """Generate an SCXML model that handles all BT related synchronization."""
```
Comment (ct2034) [inline src/as2fm/scxml_converter/bt_converter.py:79]:
```suggestion
    """Generate an SCXML model that handles all BT-related synchronization."""
```

**Review by ct2034 — CHANGES_REQUESTED** [review]

**Review by ct2034 — APPROVED** [review]

**Pull request merged**